src/ui/components/app/exporters/googleSheets/SpreadsheetExporter.vue

```diff
@@ -89,8 +90,20 @@ export default defineComponent({
       }
     };
 
+    const submit = async () => {
+      const isNewSheet = exporter.value.options.spreadsheetId.length < 30;
+      if (isNewSheet) {
+        status.value = Status.LOADING;
+        const spreadsheetId = await createSpreadsheet(exporter.value.options.spreadsheetId, exporter.value.options.credentials);
+        exporter.value.options.spreadsheetId = spreadsheetId;
+        status.value = Status.LOGGED_IN;
+      }
+      return dataToReturn.submit();
+    };
+
```

Comment:
I did almost the same code (because this is not a complicated one), but I did it in the `SheetsCombobox`. Now I think your code is better, only because I needed to manually call the `events` and I don't like that (although it should be OK if you're creating your own input component)

```js
const createNewSpreadsheet = async () => {
  if (isNewSpreadsheet.value && auth.value) {
    creating.value = true;
    const newSpreadsheet = await createSpreadhseet(props.value!, auth.value).then(({ data }) => data);
    creating.value = false;
    userSpreadsheets.value.push({ id: newSpreadsheet.spreadsheetId, name: newSpreadsheet.properties?.title });
    listeners.input?.(newSpreadsheet.spreadsheetId);
    listeners.change?.(newSpreadsheet.spreadsheetId);
  }
};
```